samples/guestbook/gcp/deploy/main.go

```diff
@@ -0,0 +1,195 @@
+		"{{database_instance}}": tfState.DatabaseInstance.Value,
+		"{{database_region}}":   tfState.DatabaseRegion.Value,
+		"{{motd_var_config}}":   tfState.MotdVarConfig.Value,
+		"{{motd_var_name}}":     tfState.MotdVarName.Value,
+	}
+	for old, new := range replacements {
+		gby = strings.Replace(gby, old, new, -1)
+	}
+	if err := ioutil.WriteFile(filepath.Join(tempDir, "guestbook.yaml"), []byte(gby), 0666); err != nil {
+		return fmt.Errorf("writing guestbook.yaml: %v", err)
+	}
+
+	// Build Guestbook Docker image.
+	log.Printf("Building %s...", imageName)
+	build := exec.Command("vgo", "build", "-o", "gcp/guestbook")
+	env := append(build.Env, "GOOS=linux", "GOARCH=amd64")
+	env = append(env, os.Environ()...)
+	build.Env = env
+	absDir, err := filepath.Abs(guestbookDir)
+	if err != nil {
+		return fmt.Errorf("getting abs path to guestbook dir (%s): %v", guestbookDir, err)
+	}
+	build.Dir = absDir
+	build.Stderr = os.Stderr
+	if err := build.Run(); err != nil {
+		return fmt.Errorf("building guestbook app by running %v: %v", build.Args, err)
+	}
+	cbs := gcp.cmd("container", "builds", "submit", "-t", imageName, filepath.Join(guestbookDir, "gcp"))
+	if out, err := cbs.CombinedOutput(); err != nil {
+		return fmt.Errorf("building container image with %v: %v: %s", cbs.Args, err, out)
+	}
+
+	// Run on Kubernetes.
+	log.Printf("Deploying to %s...", tfState.ClusterName.Value)
+	getCreds := gcp.cmd("container", "clusters", "get-credentials", "--zone", zone, tfState.ClusterName.Value)
+	getCreds.Stderr = os.Stderr
+	if err := getCreds.Run(); err != nil {
+		return fmt.Errorf("getting credentials with %v: %v", getCreds.Args, err)
+	}
+	kubeCmds := [][]string{
+		{"kubectl", "apply", "-f", filepath.Join(tempDir, "guestbook.yaml")},
+		// Force pull the latest image.
+		{"kubectl", "scale", "--replicas", "0", "deployment/guestbook"},
+		{"kubectl", "scale", "--replicas", "1", "deployment/guestbook"},
+	}
+	for _, kcmd := range kubeCmds {
+		cmd := exec.Command(kcmd[0], kcmd[1:]...)
+		if out, err := cmd.CombinedOutput(); err != nil {
+			return fmt.Errorf("running %v: %v: %s", cmd.Args, err, out)
+		}
+	}
+
+	// Wait for endpoint then print it.
+	log.Printf("Waiting for load balancer...")
+	for {
+		outb, err := runb("kubectl", "get", "service", "guestbook", "-o", "json")
+		if err != nil {
+			return err
+		}
+		var t thing
+		if err := json.Unmarshal(outb, &t); err != nil {
+			return fmt.Errorf("parsing JSON output: %v", err)
+		}
+		i := t.Status.LoadBalancer.Ingress
+		if len(i) == 0 || i[0].IP == "" {
+			dt := time.Second
+			log.Printf("No ingress returned in %s. Trying again in %v", outb, dt)
+			time.Sleep(dt)
+			continue
+		}
+		endpoint := i[0].IP
+		log.Printf("Deployed at http://%s:8080", endpoint)
+		break
+	}
+	return nil
+}
+
+type thing struct{ Status *s }
+type s struct{ LoadBalancer lb }
+type lb struct{ Ingress []ing }
+type ing struct{ IP string }
+
+type gcloud struct {
+	// project ID
```

Comment:
The field should be called `projectID` to remove the need for this comment.

Reply:
Done.